Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -201,290 +226,162 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
+
+            // update the target with a lastPublished date
+            const updatedPublishTargets = setting.publishTargets.map((profile) => {
+              if (profile.name === selectedTarget) {
+                return {
+                  ...profile,
+                  lastPublished: new Date(),
+                };
+              } else {
+                return profile;
+              }
+            });
+
+            await setPublishTargets(updatedPublishTargets, projectId);
+          }
         }
+      }
+      setBotStatusList(
+        botStatusList.map((bot) => {
+          const item = items.find((i) => i.id === bot.id);
+          if (item) {
+            item.status = 202;
+            return item;
+          }
+          return bot;
+        })
       );
-
-      if (result) {
-        if (settings.publishTargets && settings.publishTargets.length > index) {
-          const targets = settings.publishTargets.slice(0, index).concat(settings.publishTargets.slice(index + 1));
-          await setPublishTargets(targets, projectId);
-          // redirect to all profiles
-          setSelectedTarget(undefined);
-          onSelectTarget('all');
-        }
+    }
+  };
+  const changePublishTarget = (publishTarget, botStatus) => {
+    const newBotStatusItems = botStatusList.map((status) => {
+      if (status.id === botStatus.id) {
+        status.publishTarget = publishTarget;
       }
-    },
-    [settings.publishTargets, projectId, botName]
-  );
+      return status;
+    });
+    setBotStatusList(newBotStatusItems);
+  };
 
   useEffect(() => {
     setCurrentPageMode('notifications');
   }, []);
 
   return (
     <Fragment>
-      <Dialog
-        dialogContentProps={dialogProps}
-        hidden={addDialogHidden}
-        minWidth={450}
-        modalProps={{ isBlocking: true }}
-        onDismiss={() => setAddDialogHidden(true)}
-      >
-        {dialogProps.children}
-      </Dialog>
-      <Dialog
-        dialogContentProps={editDialogProps}
-        hidden={editDialogHidden}
-        minWidth={450}
-        modalProps={{ isBlocking: true }}
-        onDismiss={() => setEditDialogHidden(true)}
-      >
-        {editDialogProps.children}
-      </Dialog>
       {!publishDialogHidden && (
-        <PublishDialog
-          projectId={projectId}
-          target={selectedTarget}
-          onDismiss={() => setPublishDialogHidden(true)}
-          onSubmit={publish}
-        />
-      )}
-      {!pullDialogHidden && (
-        <PullDialog projectId={projectId} selectedTarget={selectedTarget} onDismiss={() => setPullDialogHidden(true)} />
+        <PublishDialog items={selectedBots} onDismiss={() => setPublishDialogHidden(true)} onSubmit={publish} />
       )}
+      {!pullDialogHidden &&
+        selectedBots.map((bot) => {
+          const selectedTarget = bot.publishTargets?.find((target) => target.name === bot.publishTarget);
+          const botProjectId = bot.id;
+          return (
```

Comment:
Should this be rendered conditional on selectedTarget as it can be undefined?

Reply:
`SelectedTarget` won't be undefined, the default value is `[]`